src/core/ext/transport/chttp2/transport/chttp2_transport.c

```diff
@@ -1623,6 +1623,7 @@ void grpc_chttp2_mark_stream_closed(grpc_exec_ctx *exec_ctx,
       remove_stream(exec_ctx, t, s->id,
                     removal_error(GRPC_ERROR_REF(error), s, "Stream removed"));
     }
+    grpc_chttp2_list_remove_waiting_for_concurrency(t, s);
```

Comment:
Maybe an else around this to make it clearer that this case is only interesting prior to stream id allocation?